docs/contribute/DEV_DOCS.md

```diff
@@ -45,55 +45,48 @@ output of `augur` installed with pip and `./bin/augur` from your local source co
 
-On the other hand, [**doctests**](https://docs.python.org/3/library/doctest.html) are a type of tests that are written within a module's docstrings.
-They can be helpful for testing a real-world example and determining if a regression is introduced in a particular module.
+Tests consist of:
 
-A pull request that contributes new code **should always contain unit tests**.
-Optionally, a pull request may also contain doctests if the contributor believes a doctest would improve the documentation and execution of a real world example.
+1. Unit tests
+2. Doctests
+3. Functional tests
 
-We test augur's command line interface with functional tests implemented with the [Cram framework](https://bitheap.org/cram/).
-These tests complement existing unit tests of individual augur Python functions by running augur commands on the shell and confirming that these commands:
-
-1. execute without any errors
-2. produce exactly the expected outputs for the given inputs
+The combined test coverage is currently [![codecov coverage badge showing % coverage - select for details](https://codecov.io/gh/nextstrain/augur/graph/badge.svg?token=n3ZS7rMRhY)](https://codecov.io/gh/nextstrain/augur). We realize that this number is not 100%, so the automated CI GitHub Actions workflow is augmented with external testing via running pathogen repo workflows using CI-specific input data.
 
-These tests can reveal bugs resulting from untested internal functions or untested combinations fo internal functions.
-
-Functional tests should either:
+#### 1. Unit tests
 
-* suitably test a single augur command with an eponymously named Cram file in `tests/functional/` (e.g., `mask.t` for augur mask)
+Unit tests are written using [pytest](https://docs.pytest.org).
+Augur's unit tests are located in the `tests` directory and prefixed with `test_`. There is generally one test file for each code file.
 
-OR
+#### 2. Doctests
 
-* test a complete build with augur commands with an appropriately named Cram file in `tests/builds/` (e.g., `zika.t` for the example Zika build)
+[Doctests](https://docs.python.org/3/library/doctest.html) are a type of test that are written within a module's docstrings.
+They can be helpful for testing a real-world example and determining if a regression is introduced in a particular module. They are run via `pytest`.
 
-##### Functional tests of specific commands
+#### 3. Functional tests
 
-Functional tests of specific commands consist of a single Cram file per test and a corresponding directory of expected inputs and outputs to use for comparison of test results.
+Augur's command line interface is tested by functional tests implemented with the [Cram framework](https://bitheap.org/cram/).
+These tests complement existing unit tests of individual augur Python functions by running augur commands on the shell and confirming that these commands:
 
-The Cram file should test most reasonable combinations of command arguments and flags.
+1. execute without any errors
```

Comment:
is this true? i was under the impression that Cram did not look at the exit code of the command, and only compared observed output to expected. (I base this on the [expected failure tests](https://github.com/nextstrain/ingest/blob/main/tests/transform-genbank-location/transform-genbank-location.t#L10-L15) I wrote recently.

Reply:
Cram inspects output code. You can specify an expected non-zero code however.

Reply:
@genehack The expected failure tests you wrote recently work because the exit code is still 0. If the exit code was non-zero, you'd have to include it in the Cram test.

Reply:
> @genehack The expected failure tests you wrote recently work because the exit code is still 0. If the exit code was non-zero, you'd have to include it in the Cram test.

ah, in that case, the code i wrote has a bug :grin: